eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventSender.java

```diff
@@ -122,30 +117,47 @@
     }
 
     private Mono<Void> sendInternal(Flux<EventData> events, EventBatchingOptions options) {
-        final String partitionId = senderOptions.partitionId();
+        final String partitionKey = options.partitionKey();
+        if (!ImplUtils.isNullOrEmpty(partitionKey) && partitionKey.length() > MAX_PARTITION_KEY_LENGTH) {
+            throw new IllegalArgumentException(
+                String.format(Locale.US, "PartitionKey exceeds the maximum allowed length of partitionKey: %s", MAX_PARTITION_KEY_LENGTH));
+        }
 
         //TODO (conniey): When we implement partial success, update the maximum number of batches or remove it completely.
         return events.collect(new EventDataCollector(options, 1))
-            .flatMap(list -> sendBatch(partitionId, Flux.fromIterable(list)));
+            .flatMap(list -> sendBatch(Flux.fromIterable(list)));
     }
 
-    private Mono<Void> sendBatch(String partitionId, Flux<EventDataBatch> eventBatches) {
+    private Mono<Void> sendBatch(Flux<EventDataBatch> eventBatches) {
```

Comment:
.NET just calls this `SendAsync`.    Is there something that the `batch` is intended to differentiate, with us no longer supporting a single message send?

Reply:
Nope. I'll go and fix that. It was remnants of my initial prototype from eons ago.